chia/pools/pool_wallet.py

```diff
@@ -471,22 +471,32 @@ async def pk_to_sk(pk: G1Element) -> PrivateKey:
                 return self.standard_wallet.secret_key_store.secret_key_for_public_key(pk)
             return owner_sk
 
-        spends = [coin_spend]
-        if fee > 0:
-            # TODO: test failure UX when fee is higher than available balance
-            get_max_send_amount = await self.standard_wallet.get_max_send_amount()
-            assert get_max_send_amount > fee
-            my_address = await self.standard_wallet.get_new_puzzlehash()
-            fee_spends = await self.standard_wallet._generate_unsigned_transaction(uint64(0), my_address, fee)
-            spends.extend(fee_spends)
         return await sign_coin_spends(
-            spends,
+            [coin_spend],
             pk_to_sk,
             self.wallet_state_manager.constants.AGG_SIG_ME_ADDITIONAL_DATA,
             self.wallet_state_manager.constants.MAX_BLOCK_COST_CLVM,
         )
 
-    async def generate_travel_transaction(self, fee: uint64) -> TransactionRecord:
+    async def generate_fee_transaction(self, fee: uint64, coin_announcements=None) -> TransactionRecord:
+        fee_tx = await self.standard_wallet.generate_signed_transaction(
+            uint64(0),
+            (await self.standard_wallet.get_new_puzzlehash()),
+            fee=fee,
+            origin_id=None,
+            coins=None,
+            primaries=None,
+            ignore_max_send_amount=False,
+            coin_announcements_to_consume=coin_announcements,
+        )
+        return fee_tx
+
+    async def publish_transactions(self, travel_tx: TransactionRecord, fee_tx: Optional[TransactionRecord]):
+        await self.wallet_state_manager.add_pending_transaction(travel_tx)
+        if fee_tx is not None:
+            await self.wallet_state_manager.add_pending_transaction(dataclasses.replace(fee_tx, spend_bundle=None))
+
+    async def generate_travel_transactions(self, fee: uint64) -> Tuple[TransactionRecord, Optional[TransactionRecord]]:
```

Comment:
Originally, I wrote this to return `List[TransactionRecord]`, but I changed it so that we could keep the output of the RPCs backwards compatible.